### PR TITLE
Private data optimization: purge transient store in background

### PR DESCRIPTION
**Pull request by cendhu**
#### Type of change

- Improvement (improvement to code, performance, etc)

#### Description

By purging entries present in the transient store in the background, we can improve the overall throughput of private data collection by **22%**

#### Related issues

sub-task https://jira.hyperledger.org/browse/FAB-17986
story https://jira.hyperledger.org/browse/FAB-17984

### CODE REVIEW

File: gossip/privdata/coordinator_test.go [hunk 1]
```diff
@@ -1726,11 +1713,13 @@ func TestPurgeBelowHeight(t *testing.T) {
 	block.Header.Number = 10
 	pvtData := pdFactory.addRWSet().addNSRWSet("ns1", "c1").create()
 	// test no blocks purged yet
-	assertPurged(false)
+	purged = false
+	require.Eventually(t, assertPurged, 2*time.Second, 100*time.Millisecond)
```
Comment (manish-sethi) [inline gossip/privdata/coordinator_test.go]: same comment as above

File: gossip/privdata/coordinator_test.go [hunk 2]
```diff
@@ -1174,18 +1163,20 @@ func TestCoordinatorStoreBlock(t *testing.T) {
...
 	fmt.Println("Scenario IV")
 	// Scenario IV: Block came with more than sufficient private data alongside it, some of it is redundant.
 	pvtData = pdFactory.addRWSet().addNSRWSet("ns1", "c1", "c2", "c3").
 		addRWSet().addNSRWSet("ns2", "c1", "c3").addRWSet().addNSRWSet("ns1", "c4").create()
 	err = coordinator.StoreBlock(block, pvtData)
-	assertPurged("tx1", "tx2")
 	assert.NoError(t, err)
 	assertCommitHappened()
+	purgedTxns = []string{"tx1", "tx2"}
+	require.Eventually(t, assertPurgedTxns, 2*time.Second, 100*time.Millisecond)
```
[the remark this answers is not in the document]
Comment (cendhu) [reply gossip/privdata/coordinator_test.go]: Agree. I was struggling but your suggestion didn't strike my mind.
`func() bool { return assertPurgedTxns("tx1", "tx2") }` simplifies everything.

[the remark this answers is not in the document]
Comment (cendhu) [reply gossip/privdata/coordinator_test.go]: Done.

File: gossip/privdata/coordinator_test.go (hunk 1 above)
Comment (cendhu) [reply gossip/privdata/coordinator_test.go]: Done.

**Review by manish-sethi — APPROVED** [review]

**Pull request merged**